main/client/src/mill/main/client/MillClientMain.java

```diff
@@ -259,4 +262,12 @@ private static int getServerProcessesLimit(String jvmHomeEncoding) {
         }
         return processLimit;
     }
+
+    private static String md5hex(String str) throws NoSuchAlgorithmException {
+        return hexArray(MessageDigest.getInstance("md5").digest(str.getBytes(StandardCharsets.UTF_8)));
+    }
+
+    private static String hexArray(byte[] arr) {
+        return String.format("%0" + (arr.length << 1) + "x", new BigInteger(1, arr));
+    }
```

Comment:
Make these public and add a small javadoc, to reuse them from `MillServerMain`.

Reply:
Done.